interface/ceed.c

```diff
@@ -339,44 +340,42 @@ int CeedInit(const char *resource, Ceed *ceed) {
-    {"CeedCompositeOperatorCreate",ceedoffsetof(Ceed, CompositeOperatorCreate)},
-    {"VectorSetArray",             ceedoffsetof(CeedVector, SetArray)},
-    {"VectorSetValue",             ceedoffsetof(CeedVector, SetValue)},
-    {"VectorGetArray",             ceedoffsetof(CeedVector, GetArray)},
-    {"VectorGetArrayRead",         ceedoffsetof(CeedVector, GetArrayRead)},
-    {"VectorRestoreArray",         ceedoffsetof(CeedVector, RestoreArray)},
-    {"VectorRestoreArrayRead",     ceedoffsetof(CeedVector, RestoreArrayRead)},
-    {"VectorDestroy",              ceedoffsetof(CeedVector, Destroy)},
-    {"ElemRestrictionApply",       ceedoffsetof(CeedElemRestriction, Apply)},
-    {"ElemRestrictionDestroy",     ceedoffsetof(CeedElemRestriction, Destroy)},
-    {"BasisApply",                 ceedoffsetof(CeedBasis, Apply)},
-    {"BasisDestroy",               ceedoffsetof(CeedBasis, Destroy)},
-    {"TensorContractApply",        ceedoffsetof(CeedTensorContract, Apply)},
-    {"TensorContractDestroy",      ceedoffsetof(CeedTensorContract, Destroy)},
-    {"QFunctionApply",             ceedoffsetof(CeedQFunction, Apply)},
-    {"QFunctionDestroy",           ceedoffsetof(CeedQFunction, Destroy)},
-    {"OperatorApply",              ceedoffsetof(CeedOperator, Apply)},
-    {"OperatorApplyJacobian",      ceedoffsetof(CeedOperator, ApplyJacobian)},
-    {"OperatorDestroy",            ceedoffsetof(CeedOperator, Destroy)}
+  foffset foffsets[] = {
+    CEED_FTABLE_ENTRY(Ceed, Error),
+    CEED_FTABLE_ENTRY(Ceed, GetPreferredMemType),
+    CEED_FTABLE_ENTRY(Ceed, Destroy),
+    CEED_FTABLE_ENTRY(Ceed, VecCreate),
+    CEED_FTABLE_ENTRY(Ceed, ElemRestrictionCreate),
+    CEED_FTABLE_ENTRY(Ceed, ElemRestrictionCreateBlocked),
+    CEED_FTABLE_ENTRY(Ceed, BasisCreateTensorH1),
+    CEED_FTABLE_ENTRY(Ceed, BasisCreateH1),
+    CEED_FTABLE_ENTRY(Ceed, TensorContractCreate),
+    CEED_FTABLE_ENTRY(Ceed, QFunctionCreate),
+    CEED_FTABLE_ENTRY(Ceed, OperatorCreate),
+    CEED_FTABLE_ENTRY(Ceed, CompositeOperatorCreate),
+    CEED_FTABLE_ENTRY(CeedVector, SetArray),
+    CEED_FTABLE_ENTRY(CeedVector, SetValue),
+    CEED_FTABLE_ENTRY(CeedVector, GetArray),
+    CEED_FTABLE_ENTRY(CeedVector, GetArrayRead),
+    CEED_FTABLE_ENTRY(CeedVector, RestoreArray),
+    CEED_FTABLE_ENTRY(CeedVector, RestoreArrayRead),
+    CEED_FTABLE_ENTRY(CeedVector, Destroy),
+    CEED_FTABLE_ENTRY(CeedElemRestriction, Apply),
+    CEED_FTABLE_ENTRY(CeedElemRestriction, Destroy),
+    CEED_FTABLE_ENTRY(CeedBasis, Apply),
+    CEED_FTABLE_ENTRY(CeedBasis, Destroy),
+    CEED_FTABLE_ENTRY(CeedTensorContract, Apply),
+    CEED_FTABLE_ENTRY(CeedTensorContract, Destroy),
+    CEED_FTABLE_ENTRY(CeedQFunction, Apply),
+    CEED_FTABLE_ENTRY(CeedQFunction, Destroy),
+    CEED_FTABLE_ENTRY(CeedOperator, Apply),
+    CEED_FTABLE_ENTRY(CeedOperator, ApplyJacobian),
+    CEED_FTABLE_ENTRY(CeedOperator, Destroy),
+    {"END", 0} // End of lookup table - used in SetBackendFunction loop
```

Comment:
Why not `{NULL, 0}`? Would make the end comparisons cleaner.

Reply:
Agreed